app/helpers/application_helper.rb

```diff
@@ -1,2 +1,61 @@
 module ApplicationHelper
+  def user_status
+    links = []
+    if logged_in?
+      links[0] = link_to 'New Event', new_event_path, class: 'nav-link'
+      links[1] = link_to 'Log out', logout_path, method: 'delete', class: 'nav-link'
+      links[2] = link_to current_user.username.to_s, user_path(current_user.id), class: 'nav-link'
+    else
+      links[0] = link_to 'Login', 'login', class: 'nav-link'
+      links[1] = link_to 'Sign Up', 'signup', class: 'nav-link'
+    end
+    links
+  end
+
+  def usernames
+    current_user.username.to_s
+  end
+
+  def users_ids
+    current_user.id
+  end
+
+  def date_format(event, new_date = nil)
+    if event.nil?
+      new_date.to_formatted_s(:long) if !new_date == nil?
+    else
+      event.date&.to_formatted_s(:long)
+    end
+  end
+
+  def events_attended
+    current_user.attended_events
+  end
+
+  def all_events_created_by_user
+    current_user.events
+  end
```

Comment:
There should be NO application logic in the views nor helpers (no Post.new, Comment.create, current_user.friends.find_by(...)...). Everything connected with data processing should be encapsulated in models and used in controllers.

In your controller, you can use 
- `@created_events = current_user.events` to lists events created by user
- `@past_events = current_user.attended_events.previous` to lists events previoulsy attendented by user **using event model scope**. 
- `@upcoming_events = current_user.attended_events.upcoming` to lists upcoming events previously attended by user **using event model scope**.

Reply:
I tried to move the two methods into the controller but since when the application boots up there won't be a logged-in user, I run into the following error.
![nill_class_error](https://user-images.githubusercontent.com/36323922/100613385-80a47480-3325-11eb-89ff-cb1e38e7a842.PNG)

Reply:
Use an `if` statement to ensure `user_signed_in?` first before calling `@past_events`

Reply:
Thank you. It worked.